### PR TITLE
Refactor kubernetes to use helm/terraform

### PR DESCRIPTION
**Pull request by jaydeluca**
Refactor to helm and terraform:
- [x] Postgres
- [x] Orders Migrations
- [x] Orders Api
- [x] Users Api
- [ ] Service mesh
- [x] Load test
- [x] SigNoz

### CODE REVIEW

File: terraform/k8s/providers.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+variable "config_context" {
```
Comment (ianarsenault) [inline terraform/k8s/providers.tf:1]: You should update this to be a [local value](https://developer.hashicorp.com/terraform/language/values/locals)


```
locals {
  config_context = "docker-desktop"
}
```

and reference `local.config_context`

File: terraform/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Terraform
+
+The cluster will be setup via helm charts that are applied via terraform
+
+
+```
+ubectl --namespace postgres port-forward $(k get pods -n postgres | grep postgres | awk '{print $1}') 5432:5432
```
Comment (ianarsenault) [inline terraform/readme.md:7]:
```suggestion
kubectl --namespace postgres port-forward $(k get pods -n postgres | grep postgres | awk '{print $1}') 5432:5432
```

File: terraform/k8s/providers.tf [hunk 3]
```diff
@@ -0,0 +1,24 @@
+locals {
+  config_context = "docker-desktop"
+}
+
+provider "kubernetes" {
+  config_context   = var.config_context
```
Comment (ianarsenault) [inline terraform/k8s/providers.tf:6]: still referencing var.config_context here

Comment (ianarsenault) [inline terraform/k8s/providers.tf:6]:
```suggestion
  config_context   = local.config_context
```

File: k8s/charts/orders-api/templates/orders-api.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  name: {{ include "orders-api.fullname" . }}-orders-api
+  labels:
+  {{- include "orders-api.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.ordersApi.type }}
+  selector:
+    app: orders-api
+  {{- include "orders-api.selectorLabels" . | nindent 4 }}
+  ports:
+	{{- .Values.ordersApi.ports | toYaml | nindent 2 -}}
```
Comment (jaydeluca) [inline k8s/charts/orders-api/templates/orders-api.yml:13]: wtf is this? 🤔

File: api/src/main/k8s.yml [hunk 5]
```diff
@@ -2,7 +2,7 @@ database:
   driverClass: org.postgresql.Driver
   user: morpheus
   password: findneo
-  url: jdbc:postgresql://postgres:5432/sre-lab
+  url: jdbc:postgresql://postgres-postgresql.postgres.svc.cluster.local:5432/postgres
```
Comment (jaydeluca) [inline api/src/main/k8s.yml:5]: at some point will want to make this pull from a secret or something instead

File: k8s/charts/orders-api/values.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+kubernetesClusterDomain: cluster.local
+ordersApi:
+  ordersApi:
```
Comment (jaydeluca) [inline k8s/charts/orders-api/values.yaml:3]: i dont like how `helmify` did this double nesting, i'll probably go back and clean that up

File: api/src/main/k8s.yml (hunk 5 above)
Comment (ianarsenault) [reply api/src/main/k8s.yml:5]: I think we could add a terraform module that sets up the k8 cluster with the needed stuff (namespaces, configmaps, secrets) and the reference those secrets in the yaml

File: k8s/charts/orders-api/values.yaml (hunk 6 above)
Comment (ianarsenault) [reply k8s/charts/orders-api/values.yaml]: ah that's gross

File: api/src/main/k8s.yml (hunk 5 above)
Comment (jaydeluca) [reply api/src/main/k8s.yml:5]: yea good call

File: k8s/charts/orders-migrations/.helmignore [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
```
Comment (jaydeluca) [inline k8s/charts/orders-migrations/.helmignore:1]: how important are these files? can't tell if they are just fluff from helmify or if it makes sense to have them throughout the code base?

**Pull request merged**